Frontend/lib/features/create_match/single_match/create_single_match_page.dart

```diff
@@ -67,8 +67,8 @@ class _CreateSingleMatchPageState extends State<CreateSingleMatchPage> {
       DateTime matchDateTime = DateTime(selectedDate.year, selectedDate.month,
           selectedDate.day, selectedTime.hour, selectedTime.minute);
 
-      final match = MatchModel(
-        id: UniqueKey().toString(),
+      final match = Match(
+        id: null,
```

Comment:
The application seems to return an error when submitting a new match with `null` as `id` value. While the model now allows this value to be `null`, Supabase doesn't appear to allow this on my end. Entering any integer here will resolve the issue (hence, the `UniqueKey().toString()` that was implemented previously), as Supabase will happily ignore this and use the correct, autogenerated ID instead. But that would not be an ideal solution.

Instead of an `.upsert`, we might want to consider implementing a database function to add new matches to the DB instead, similar to what I'm doing for the [updated tournament page](https://github.com/bryandijkhuizen/dartproject-s7/blob/37cd1ce13059511ec19d0e3770edc210036cc97d/supabase/migrations/20240606084956_add_create_tournament_function.sql). 

<img width="907" alt="image" src="https://github.com/bryandijkhuizen/dartproject-s7/assets/11412480/2fcdf0b8-3b9f-427f-93fa-2741b4f4d9a5">